src/me/remie/vulcan/leaguetasks/task/tasks/crafting/CraftLeatherBody.java

```diff
@@ -0,0 +1,173 @@
+import simple.hooks.filters.SimpleSkills;
+import simple.hooks.queries.SimpleEntityQuery;
+import simple.hooks.wrappers.SimpleGroundItem;
+import simple.hooks.wrappers.SimpleItem;
+import simple.hooks.wrappers.SimpleNpc;
+import simple.robot.api.ClientContext;
+
+import java.util.Map;
+
+/**
+ * Created by Pepsiplaya on Mar 05, 2024 at 10:51 PM
+ */
+public class CraftLeatherBody extends LeagueTask {
+    BankManager bankManager = new BankManager(script);
+    private static final int LEATHER = 1741;
+    private static final int NEEDLE_ID = 1733;
+    private static final int THREAD_ID = 1734;
+    private static final ItemDetails NEEDLE = new ItemDetails("Needle", NEEDLE_ID, 5_000);
+    private static final ItemDetails THREAD = new ItemDetails("Thread", THREAD_ID, 5_000);
+    private final ShopHelper shopHelper;
+
+    public CraftLeatherBody(final LeagueScript script) {
+        super(script, "Craft a Leather Body");
+        this.shopHelper = new ShopHelper(this);
+        setSkillRequirement(new SkillRequirement(script.ctx, Map.of(SimpleSkills.Skills.CRAFTING, 14)));
+    }
+
+    @Override
+    public void run() {
+        if (hasItemInInventory(NEEDLE_ID) && hasItemInInventory(THREAD_ID) && hasItemInInventory(LEATHER)) {
+            if (!ctx.dialogue.dialogueOpen()) {
+                SimpleItem leather = ctx.inventory.populate().filter(LEATHER).next();
+                SimpleItem needle = ctx.inventory.populate().filter(NEEDLE_ID).next();
+                SimpleItem thread = ctx.inventory.populate().filter(THREAD_ID).next();
+                needle.menuAction("Use");
+                ctx.sleep(600);
+                leather.click(0);
+                ctx.sleep(600);
+            } else {
+                ctx.dialogue.clickDialogueOption(1);
+                ctx.sleep(600);
+            }
+        }
+        if (!hasItemInInventory(NEEDLE_ID)) {
+            shopHelper.buyItem(NEEDLE, ShopType.GENERAL_STORE);
+        }
+
+        if (!hasItemInInventory(THREAD_ID)) {
+            shopHelper.buyItem(THREAD, ShopType.GENERAL_STORE);
+        }
+
+        if (!hasItemInInventory(LEATHER)) {
+            obtainLeather();
+        }
+
+        // Proceed with crafting the leather body if all items are now obtained...
+    }
+
+    private boolean hasItemInInventory(int itemId) {
+        return !ctx.inventory.populate().filter(itemId).isEmpty();
+    }
+
+    private void obtainLeather() {
+        if (hasItemInInventory(THREAD_ID) && hasItemInInventory(NEEDLE_ID)) {
+            // Check inventory for cowhides
+            final int COWHIDE = 1739;
+            if (ctx.inventory.populate().filter(COWHIDE).population() < 1) {
+                // Teleport to cows, kill them, and loot cowhides
+                killCowsAndLootHides();
+            }
+
+            // Check if we have cowhides to tan
+            if (ctx.inventory.populate().filter(COWHIDE).population() >= 1) {
```

Comment:
We can probably remove the check for cowhide here, because if we add return above, than we know that we have when it reaches this code block.